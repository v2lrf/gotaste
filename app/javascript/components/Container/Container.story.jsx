import React, { Fragment } from 'react'
import { storiesOf } from '@storybook/react'

import { Container } from '.'

storiesOf('Container', module).add('Default', () => (
  <Fragment>
    <Container>I have a fixed width and horizontal auto margin</Container>
  </Fragment>
))
