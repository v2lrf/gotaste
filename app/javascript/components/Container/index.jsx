import React from 'react'
import PropTypes from 'prop-types'

export function Container({ children }) {
  return <div className="container mx-auto px-4">{children}</div>
}

Container.propTypes = {
  children: PropTypes.node.isRequired
}

export default Container
