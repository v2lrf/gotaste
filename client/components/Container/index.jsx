import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

function Container({ children, narrow }) {
  const containerClasses = classNames('container mx-auto px-4', {
    'max-w-xl': narrow
  })
  return <div className={containerClasses}>{children}</div>
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  narrow: PropTypes.bool
}

Container.defaultProps = {
  narrow: false
}

export default Container
