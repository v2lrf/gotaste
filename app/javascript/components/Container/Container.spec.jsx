import React from 'react'
import Container from './Container'

describe('Container', () => {
  let component
  beforeEach(() => {
    component = shallow(<Container>I know kung fu.</Container>)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
