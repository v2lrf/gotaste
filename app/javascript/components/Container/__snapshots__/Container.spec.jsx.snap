// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Snapshots should match the snapshot 1`] = `
<div
  className="container mx-auto px-4"
>
  I know kung fu.
</div>
`;
